commit 24ded29ff63d0adee798936ad20f21befcf8187b
Author: Wayne Davison <wayned@samba.org>
Date:   Sun Mar 16 06:56:26 2008 -0700

    Fixed a hang when using --remove-source-files in dry-run mode.

diff --git a/generator.c b/generator.c
index b1634bd..4ec0ac4 100644
--- a/generator.c
+++ b/generator.c
@@ -1856,7 +1856,7 @@ static void recv_generator(char *fname, struct file_struct *file, int ndx,
 		rprintf(FINFO, "generating and sending sums for %d\n", ndx);
 
   notify_others:
-	if (remove_source_files && !delay_updates && !phase)
+	if (remove_source_files && !delay_updates && !phase && !dry_run)
 		increment_active_files(ndx, itemizing, code);
 	if (inc_recurse && !dry_run)
 		cur_flist->in_progress++;
